gitlab-status/handler_test.go

```diff
@@ -73,47 +76,264 @@ func Test_appendParameters(t *testing.T) {
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.title, func(t *testing.T) {
+			logsURL, _ := FormatLogsURL(test.gatewayURL, test.event)
+			if logsURL != test.expectedURL {
+				t.Errorf("expected URL: %s got: %s", test.expectedURL, logsURL)
+			}
+		})
+	}
+}
+
+func Test_FormatEndpointURL(t *testing.T) {
+	tests := []struct {
+		title       string
+		event       *sdk.Event
+		gatewayURL  string
+		expectedURL string
+	}{
+		{
+			title: "gateway_public_url is set and object is set",
+			event: &sdk.Event{
+				Service: "exampleservicename",
+				Owner:   "exampeuser",
+			},
+			gatewayURL:  "https://cloud.o6s.io/",
+			expectedURL: "https://exampeuser.o6s.io/exampleservicename",
+		},
+		{
+			title:       "gateway_public_url is set but object is empty",
+			event:       &sdk.Event{},
+			gatewayURL:  "https://cloud.o6s.io",
+			expectedURL: "https://.o6s.io/",
+		},
+		{
+			title: "gateway_public_url is unset but object is set",
+			event: &sdk.Event{
+				Service: "exampleservicename",
+				Owner:   "exampeuser",
+			},
+			gatewayURL:  "",
+			expectedURL: "exampeuser/exampleservicename",
+		},
+		{
+			title:       "gateway_public_url is empty and object is empty",
+			event:       &sdk.Event{},
+			gatewayURL:  "",
+			expectedURL: "/",
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.title, func(t *testing.T) {
+			endpointURL, _ := FormatEndpointURL(test.gatewayURL, test.event)
+			if endpointURL != test.expectedURL {
+				t.Errorf("Expected URL: %s got: %s", test.expectedURL, endpointURL)
+			}
+		})
+	}
+}
+
+func Test_FormatDashboardURL(t *testing.T) {
+	tests := []struct {
+		title       string
+		event       *sdk.Event
+		gatewayURL  string
+		expectedURL string
+	}{
+		{
+			title: "gateway_public_url is set and object is set",
+			event: &sdk.Event{
+				Owner: "exampleowner",
+			},
+			gatewayURL:  "https://cloud.o6s.io/",
+			expectedURL: "https://system.o6s.io/dashboard/exampleowner",
+		},
+		{
+			title: "gateway_public_url is unset and object is set",
+			event: &sdk.Event{
+				Owner: "exampleowner",
+			},
+			gatewayURL:  "",
+			expectedURL: "system/dashboard/exampleowner",
+		},
+		{
+			title:       "gateway_public_url is unset and object is empty",
+			event:       &sdk.Event{},
+			gatewayURL:  "",
+			expectedURL: "system/dashboard/",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.title, func(t *testing.T) {
+			dashboardURL, _ := FormatDashboardURL(test.gatewayURL, test.event)
+			if dashboardURL != test.expectedURL {
+				t.Errorf("Expected URL: %s got: %s", test.expectedURL, dashboardURL)
+			}
+		})
+	}
+}
+
+func Test_FormatSystemURL(t *testing.T) {
+	tests := []struct {
+		title       string
+		gatewayURL  string
+		expectedURL string
+	}{
+		{
+			title:       "gateway_public_url is set with slash",
+			gatewayURL:  "https://cloud.o6s.io/",
+			expectedURL: "https://system.o6s.io",
+		},
+		{
+			title:       "gateway_public_url is set without slash",
```

Comment:
without slash where? How about `trailing slash`? trailing = at end

Reply:
Yep good proposal will add that

Reply:
Fixed that